sdk/eventhub/event-hubs/src/eventHubSender.ts

```diff
@@ -557,85 +556,18 @@ export class EventHubSender extends LinkEntity {
     const abortSignal: AbortSignalLike | undefined = options.abortSignal;
     const sendEventPromise = () =>
       new Promise<void>(async (resolve, reject) => {
-        let waitTimer: any;
-
-        let onRejected: Func<EventContext, void>;
-        let onReleased: Func<EventContext, void>;
-        let onModified: Func<EventContext, void>;
-        let onAccepted: Func<EventContext, void>;
-        let onAborted: () => void;
-
         const rejectOnAbort = () => {
           const desc: string =
-            `[${this._context.connectionId}] The send operation on the Sender "${
-              this.name
-            }" with ` + `address "${this.address}" has been cancelled by the user.`;
+            `[${this._context.connectionId}] The send operation on the Sender "${this.name}" with ` +
+            `address "${this.address}" has been cancelled by the user.`;
           log.error(desc);
           return reject(new AbortError("The send operation has been cancelled by the user."));
         };
 
         if (abortSignal && abortSignal.aborted) {
           // operation has been cancelled, so exit quickly
-          return rejectOnAbort();
-        }
-
-        onAborted = () => {
-          removeListeners();
           rejectOnAbort();
```

Comment:
You'll want to add a return here or else the function will continue running.

Reply:
`rejectOnAbort` has a `return reject(..)` in it, would we still need another `return` here?

Reply:
Yes. `rejectOnAbort` is a function. The `return` in it only returns from that function and not the outer code set of {}

Reply:
Updated..